### PR TITLE
xcat: init at 1.2.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
xcat is an advanced tool for exploiting XPath injection vulnerabilities,
featuring a comprehensive set of features to read the entire file being
queried as well as other files on the filesystem, environment variables
and directories.

https://github.com/orf/xcat

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 118142` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.xpath-expressions</li>
    <li>python39Packages.xpath-expressions</li>
    <li>xcat</li>
  </ul>
</details>

File: pkgs/development/python-modules/xpath-expressions/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  disabled = pythonOlder "3.5";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0yn2n733cffk2pa7m1g3bp2kq8v9fa6dh91rxssly2sbpng0bk83";
+  };
+
+  nativeBuildInputs = [
+    poetry
```
Comment (FRidh) [inline pkgs/development/python-modules/xpath-expressions/default.nix:20]: `poetry-core`. If it does not work with only `poetry-core`, then upstream needs to fix their dependency.

File: pkgs/tools/security/xcat/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  src = fetchFromGitHub {
+    owner = "orf";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1y809dhfxb6ddzn519ci5cd0yrhl6s2p1xk9nfbi8wk5djzqlldw";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [
+    poetry
```
Comment (FRidh) [inline pkgs/tools/security/xcat/default.nix:20]: same

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118142` at f928049c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.xpath-expressions</li>
    <li>python39Packages.xpath-expressions</li>
    <li>xcat</li>
  </ul>
</details>

**Pull request merged**